### ISSUE TITLE
Deprecated criterion 4.1.1 Parsing

### ISSUE THREAD

**Issue by LaZay** [w3c/wcag#3641]
Providing a well formed, and valid HTML is a requirement to ensure the quality of user experience. As a matter of fact, it is a garantee that the DOM tree is **complete** and fully readable by any screen reader, but also by any browser, and reading system (i. e. reading EPUBs, containing HTML files).

There is still a bunch of bullshit HTML around, though it has dramatically dicreased these past years, thanks to the W3C validator, and in fact thanks the WCAG 4.1.1 requirement. Allowing badly formed, or invalid HTML will lead to a catastrophic user experience (keyboard navigation on badly formed / invalid headings typically).

Deprecating this former A criterion is going to be at major risk for every users (not only those using ATs). Contrary to what is said in the [WCAG 4.1.1 note](https://www.w3.org/TR/WCAG22/#parsing), assistive technologies are not the only technologies concerned with this criterion. It cannot be the only reason for deprecation. Do we need to make a clear distinction between HTML parsing versus HTML quality control? If yes, maybe the 4.1.1 Parsing should be reintroduced, but renamed as quality control. The [W3C validator](https://validator.w3.org/) is supposed to be THE free, and reliable tool to achieve HTML quality control. Is their any problem with its maintenance?

**Comment by patrickhlauke** [comment]
> There is still a bunch of bullshit HTML around

and the HTML specification itself includes specific algorithms for how parsers/rendering engines can turn the bullshit HTML into a consistent DOM tree - which was the problem at the time 4.1.1 was devised, where different browsers would output wildly different DOM trees as they used their own secret error correction approaches. Now, if bullshit HTML leads to a malformed DOM tree, it will do so the same way in all compliant modern browsers, meaning you can then consistently test for its actual detrimental effects.

> Allowing badly formed, or invalid HTML will lead to a catastrophic user experience (keyboard navigation on badly formed / invalid headings typically)

which will fail other, more specific success criteria

**Comment by LaZay** [comment]
I understand your point. But:
- No one knows what he doesn't know. Accessibility QA is not achieved by the editors. If some content disappears, no one wil notice.
- HTML validation is a very simple and reliable mean to check the HTML robustness criterion. Contrarly to almost every other functional criterions, whose technical sufficent tests, and failures are much difficult to master. It is super useful to add such a requirement in HTML, when subcontracting born accessible HTML structuring.

**Comment by kimviens** [comment]
I don't think that the removal of 4.1.1 Parsing makes it that web coders should not check their HTML using validator. I too was worried by the removal of this criteria but I have now discovered that the issues caused by bad HTML are mostly either 1.3.1 Info and Relationships or 4.1.2 Name, Role, Value problems.

4.1.1 Parsing was causing false positive reports probably because: 
- Not all parsing issues were accessibility issues but all accessibility issues found under 4.1.1 Parsing could be attributed to other criterion depending on their effects. 

So 4.1.1 Parsing is not needed for filing issues but the HTML validator can still be used in testing. However, each error that the validator finds should be tested for real impact on accessibility.

Finally: Its super rare to have a Content Management System (CMS) to not have at least 1 parsing issue. Making it super difficult to achieve for those softwares.

**Comment by LaZay** [comment]
I am still not totally convinced that wrong HTML will ALWAYS have side effects on 1.3.1 and 4.1.2  in every tester/user environments. But I might be wrong.

The time for early adopters, and engaged people is over. Accessibility is going to be an obligation in Europe in June 2024. People who make contents and web apps accessible by obligation only will do the less they can, just to be conformant (whatever the user experience will be). Furthermore, if the criterion disapears, well-formed and valid HTML will not be checked during **accessibility audit** anymore.

**Comment by mbgower** [comment]
> I am still not totally convinced that wrong HTML will ALWAYS have side effects on 1.3.1 and 4.1.2 in every tester/user environments

If wrong HTML does not have a side effect on 1.3.1 or 4.1.2, then it's less likely to cause an accessibility issue, getting back to a motivator for why this was removed from WCAG 2.2. And remember, there are other criteria it could also fail.

> Furthermore, if the criterion disapears, well-formed and valid HTML will not be checked during accessibility audit anymore.

Again, I question the validity of this statement. Do you know of any maker of an accessibility test engine who is _removing_ existing checks as a result of this change? Or are they like IBM, merely reporting issues flagged by an existing rule against another criterion? There is a potential some may downgrade a  WCAG 2.1 rule that maps ONLY to 4.1.1 (say from a failure to a caution), but that's the most I'm aware of.

**Comment by patrickhlauke** [comment]
> If wrong HTML does not have a side effect on 1.3.1 or 4.1.2, then it's less likely to result in an accessibility issue for users

I think the point was "in every tester/user environment", but yes the assumption that underlies the deprecation of 4.1.1 is that now we have consistent parsing/turning things into a consistent DOM in browsers/user agents that conform to HTML5/living standard.

**Comment by LaurenceZaysser** [comment]
Unfortunatly, in the publishing field, tools and processes remain crazy (amateurs, not professional at all). I am afraid that this can apply to other sectors of activity, since digital is not managed by computer scientist anymore. Millennials cost nothing + who cares if they are only digital users (instead of conceptors/developers). Things have not progressed these last 25 past years. On the contrary, they have regressed ! 
HELP!

**Comment by patrickhlauke** [comment]
As I don't believe the decision to deprecate 4.1.1 will be reopened/reversed, I'm going to close this.
